### PR TITLE
Support polymesh schema parsing

### PR DESCRIPTION
**Pull request by zenith1994**

### CODE REVIEW

**Comment by zenith1995** [comment]
Hello. I've read your message. Could you explain your requirements? I want
to contribute to your library. I'm implementing writing feature of alembic
files.

File: src/schemas/polymesh_schema.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+#[derive(Debug)]
+pub struct PolymeshSchema {
+    base_geom: BaseGeomSchema,
+    pub facecounts: ArrayPropertyReader,
+    pub faceindices: ArrayPropertyReader,
+    pub vertices: ArrayPropertyReader,
+    pub normals: Option<ArrayPropertyReader>,
+    pub uv: Option<CompoundPropertyReader>,
+    // pub facesets: Option<ArrayPropertyReader>,
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:18]: Can this commented code be removed?

File: src/schemas/polymesh_schema.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    pub velocities: Option<ArrayPropertyReader>
+}
+
+impl PolymeshSchema {
+    pub fn new_from_object_reader(
+        object: &ObjectReader,
+        reader: &mut dyn ArchiveReader,
+        archive: &Archive,
+    ) -> Result<Self> {
+
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:28]: Rustfmt should delete this empty line?

File: src/schemas/polymesh_schema.rs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    }
+    pub fn has_uv(&self) -> bool {
+        self.uv.is_some()
+    }
+    // pub fn has_facesets(&self) -> bool {
+    //     self.facesets.is_some()
+    // }
+    pub fn has_velocities(&self) -> bool {
+        self.velocities.is_some()
+    }
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:103]: Keep empty lines between the functions, and remove or un-comment commented-out code :)

**Review by MarijnS95 — COMMENTED** [review]
Cool stuff, thanks for implementing this missing part!

**Comment by zenith1995** [comment]
I did it.

File: src/schemas/polymesh_schema.rs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+use crate::object_reader::ObjectReader;
+use crate::pod::*;
+use crate::property::*;
+use crate::reader::ArchiveReader;
+use crate::result::*;
+use crate::Archive;
+pub use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct PolymeshSchema {
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:11]: Nit: Rename to `PolyMeshSchema` to match the schema name (notice capitalized `M`)?

File: src/schemas/polymesh_schema.rs [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+
+    pub fn has_normals(&self) -> bool {
+        self.normals.is_some()
+    }
+    pub fn has_uv(&self) -> bool {
+        self.uv.is_some()
+    }
+    pub fn has_velocities(&self) -> bool {
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:87]:
```suggestion
    }

    pub fn has_uv(&self) -> bool {
        self.uv.is_some()
    }

    pub fn has_velocities(&self) -> bool {
```

**Comment by zenith1994** [comment]
Please check.

**Review by Jasper-Bekkers — APPROVED** [review]

File: src/schemas/polymesh_schema.rs [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+        // load optional properties
+        let normals = properties
+            .load_sub_property_by_name_checked("N", reader, archive, Some(&F32X3_TYPE))?
+            .map(|x| x.try_into())
+            .transpose()?;
+
+        let uv = properties
+            .load_sub_property_by_name("uv", reader, archive)?
+            .map(|x| -> Result<CompoundPropertyReader> { Ok(x.try_into()?) })
```
Comment (MarijnS95) [inline src/schemas/polymesh_schema.rs:61]: Hint: /notSince the compiler can figure out what the resulting type is, we can actually simplify this to:
```suggestion
            .map(|x| x.try_into())
```

And we should do the same in `xform_Schema` where it was doing the same, though other files have also simplified this :)

**Review by MarijnS95 — APPROVED** [review]
Thanks for making the rename change, the rest looks good!

**Pull request merged**